core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -47,7 +47,7 @@ class FsHistoryProviderSuite extends SparkFunSuite with BeforeAndAfter with Matc
   private var testDir: File = null
 
   before {
-    testDir = Utils.createTempDir()
+    testDir = Utils.createTempDir(namePrefix = s"a b%20c+d")
```

Comment:
Hm, because this affects the whole suite, I wonder if it's more sensible to test this behavior once in a specific test case, rather than test the behavior everywhere. It works, and I don't have a big problem with it, just looks a little surprising to read.